zebra-state/src/service.rs

```diff
@@ -1554,6 +1557,60 @@ impl Service<ReadRequest> for ReadStateService {
                 .map(|join_result| join_result.expect("panic in ReadRequest::BestChainBlockHash"))
                 .boxed()
             }
+
+            // Used by get_block_template RPC.
+            #[cfg(feature = "getblocktemplate-rpcs")]
+            ReadRequest::ChainInfo() => {
+                metrics::counter!(
+                    "state.requests",
+                    1,
+                    "service" => "read_state",
+                    "type" => "chain_info",
+                );
+
+                let timer = CodeTimer::start();
+
+                let state = self.clone();
+                let network = self.network;
+
+                // # Performance
+                //
+                // Allow other async tasks to make progress while concurrently reading blocks from disk.
+                let span = Span::current();
+                tokio::task::spawn_blocking(move || {
+                    span.in_scope(move || {
+                        // Tip
+                        let tip = state.non_finalized_state_receiver.with_watch_data(
+                            |non_finalized_state| {
+                                read::tip(non_finalized_state.best_chain(), &state.db)
+                            },
+                        );
+
+                        // Difficulty
+                        let expected_difficulty = tip.map(|tip| {
+                            state.non_finalized_state_receiver.with_watch_data(
+                                |non_finalized_state| {
```

Comment:
This could be part of the `with_watch_data` call above or with `latest_non_finalized_state`

Reply:
Thanks, this was done at https://github.com/ZcashFoundation/zebra/pull/5659/commits/9a12c17587376aab9220e331ea4accebfa17d7d0